src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+}
+TEST_F(StressCapiConfigChanges, GetMetricsDuringLoad) {
+    bool performWholeConfigReload = false;                        // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::testCurrentRequestsMetric,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemoveDefaultVersionDuringPredictLoad) {
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET,  // we hit default version which is unloaded already but default is not changed yet
+        StatusCode::MODEL_VERSION_MISSING};              // there is no default version since all are either not loaded properly or retired
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE};
+    // we need whole config reload since there is no other way to dispose
+    // all model versions different than removing model from config
+    bool performWholeConfigReload = true;
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangeToShapeAutoDuringPredictLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::changeToAutoShape,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemovePipelineDefinitionDuringPredictLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_ANYMORE};  // we expect to stop creating pipelines
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::removePipelineDefinition,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangedPipelineConnectionNameDuringPredictLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
```

Comment:
allowed should be either empty or StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET I think